ext/datadog_profiling_native_extension/stack_recorder.c

```diff
@@ -571,6 +589,14 @@ static VALUE _native_serialize(DDTRACE_UNUSED VALUE _self, VALUE recorder_instan
 
   ddog_prof_EncodedProfile_drop(&serialized_profile.ok);
 
+  // It's now OK to again do things that can trigger exceptions
+
+  // @ivoanjo: Maybe this could be done without the GVL being held?
+  ddog_prof_MaybeError result = ddog_prof_ManagedStringStorage_advance_gen(state->string_storage);
```

Comment:
If the string storage itself is threadsafe then we should?

Reply:
I left the comment as it looked slightly awkward to plumb the action to happen outside the GVL and to bring its result back in afterwards but I tried it now and it seems reasonable so outside of the GVL it goes in 1947ea0752fa3663f6fe238fb7bed77ba0ce4345 .